server/src/main/java/org/cloudfoundry/identity/uaa/oauth/OauthGrant.java

```diff
@@ -1,21 +1,20 @@
 package org.cloudfoundry.identity.uaa.oauth;
 
-public enum OauthGrant {
-    JWT_BEARER("jwt_bearer"),
-    CLIENT_CREDENTIALS("client_credentials"),
-    PASSWORD("password"),
-    IMPLICIT("implicit"),
-    AUTHORIZTION_CODE("authorization_code"),
-    REFRESH_TOKEN("refresh_token")
-    ;
-    
-    private final String value;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class OauthGrant {
     
-    OauthGrant(String value) {
-        this.value = value;
-    }
+    public static final String CLIENT_CREDENTIALS = "client_credentials";
+    public static final String PASSWORD = "password";
+    public static final String IMPLICIT = "implicit";
+    public static final String AUTHORIZATION_CODE = "authorization_code";
+    public static final String REFRESH_TOKEN = "refresh_token";
+    public static final String JWT_BEARER = "urn:ietf:params:oauth:grant-type:jwt-bearer";
     
-    public String value() {
-        return this.value;
-    }
- }
```

Comment:
//Alternatively as a matter of preference ...sticking with enum could be more readable and compiler friendly...
   
    private static final OauthGrant[] copyOfValues = values();

    public static OauthGrant forName(String name) {
        for (OauthGrant value : copyOfValues) {
            if (value.name().equals(name)) {
                return value;
            }
        }
        return null;
    }